common/src/main/java/com/wynntils/models/worlds/WorldStateModel.java

```diff
@@ -147,18 +153,20 @@ public void update(PlayerDisplayNameChangeEvent e) {
         Component displayName = e.getDisplayName();
         StyledText name = StyledText.fromComponent(displayName);
         Matcher m = name.getMatcher(WORLD_NAME);
-        if (setWorldIfMatched(m)) return;
+        if (setWorldIfMatched(m, false)) return;
         // must check in this order as housing name regex matches anything that WORLD_NAME would match, housing names
         // need to exclude world names.
         Matcher housingNameMatcher = name.getMatcher(HOUSING_NAME);
-        setWorldIfMatched(housingNameMatcher);
+        setWorldIfMatched(housingNameMatcher, true);
     }
 
-    private boolean setWorldIfMatched(Matcher m) {
+    private boolean setWorldIfMatched(Matcher m, boolean housing) {
         if (m.find()) {
-            String worldName = m.group(1);
+            String worldName = housing ? currentWorldName : m.group(1);
```

Comment:
This assumes that even when joining other worlds by joining houses, you get to see the updated world name for a small bit. I only assume this is correct because of the duplicate messages this PR aims to fix, which I assume is triggered by first, the world change, then by the the world changing to "housing".

Reply:
No, this doesn't fix the changing worlds housing issue, only it changing the world name to the name of the housing, which then makes features that use current world name not work correctly. Server uptime overlay, bomb bell command/overlay, discord rich presence etc. 

So since `setState` checks the world names aren't equal before posting the world change event, it won't incorrectly post the event as the `newWorldName` is still set to `currentWorldName`

Reply:
Okay, so it's not a full fix, but at least it patches the issue. Sure.